### PR TITLE
feat: added user menu component

### PR DESCRIPTION
**Pull request by CzarekDryl**
image
image

Adding popup user menu component.

### CODE REVIEW

**Comment by KatarzynaStefanowiczPagepro** [comment]
Maybe we can standardize this component? I mean we have a box (something like a Card) with shadow and content here.
Please, check the component User Avatar Popover in the storybook that is existing. We are using there a similar base component. It could be done in separate pull requests.

File: src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react';
+import clsx from 'clsx';
+import Icon from '~shared/Icon';
+import styles from './ThemeSwitcher.module.css';
+import { formatMessage } from '~utils/yup/tests/helpers';
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx:5]: import { useIntl } from 'react-intl';
  const { formatMessage } = useIntl();

File: src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+const ThemeSwitcher = () => {
+  const [isLightMode, setIsLightkMode] = useState<boolean>(true);
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx:8]: type explicitly is not necessary, just useState(true)

File: src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  return (
+    <button
+      type="button"
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx:12]: missing aria-label

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+              <p className="ml-2">
+                <FormattedMessage {...MSG.getStartedTitle} />
+              </p>
+            </Link>
+          </li>
+          <li className="mb-4">
+            <Link to="/" className="flex items-center">
+              <Icon name="lifebuoy" appearance={{ size: 'tiny' }} />
+              <p className="ml-2">
+                <FormattedMessage {...MSG.contactAndSupportTitle} />
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:68]: the same

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+              </p>
+            </Link>
+          </li>
+          <li className="mb-4">
+            <Link to="/" className="flex items-center">
+              <Icon name="code" appearance={{ size: 'tiny' }} />
+              <p className="ml-2">
+                <FormattedMessage {...MSG.developersTitle} />
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:76]: the same

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+              </p>
+            </Link>
+          </li>
+          <li>
+            <Link to="/" className="flex items-center">
+              <Icon name="briefcase" appearance={{ size: 'tiny' }} />
+              <p className="ml-2">
+                <FormattedMessage {...MSG.legalAndPrivacyTitle} />
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:84]: the same

File: src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.module.css [hunk 7]
```diff
@@ -0,0 +1,5 @@
+.themeSwitcherSpan {
+  @apply h-[1.75rem] rounded flex items-center justify-center shadow-content absolute left-[0.375rem] top-[0.375rem] transition-all duration-300 z-0;
```
[the remark this answers is not in the document]
Comment (CzarekDryl) [reply src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.module.css]: Changed duration

File: src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.module.css [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+
+  width: calc(50% - 0.5rem);
```
[the remark this answers is not in the document]
Comment (CzarekDryl) [reply src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.module.css]: Changed

File: src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react';
+import clsx from 'clsx';
+import Icon from '~shared/Icon';
+import styles from './ThemeSwitcher.module.css';
+import { formatMessage } from '~utils/yup/tests/helpers';
+
```
[the remark this answers is not in the document]
Comment (CzarekDryl) [reply src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx:6]: Added

File: src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+const ThemeSwitcher = () => {
```
[the remark this answers is not in the document]
Comment (CzarekDryl) [reply src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx]: FC added

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+import { UserMenuProps } from './types';
+import { useAppContext, useMobile } from '~hooks';
+import Button from '~shared/Extensions/Button';
+import { MSG } from './consts';
+import Link from '~shared/Link';
+import Icon from '~shared/Icon';
+import ThemeSwitcher from '~common/Extensions/ThemeSwitcher';
+import styles from './UserMenu.module.css';
+
+const UserMenu: FC<UserMenuProps> = ({ tooltipProps, setTooltipRef, isWalletConnected }) => {
```
[the remark this answers is not in the document]
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx:14]: Added

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx]: Changed to formatMessage

File: src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx (hunk 1 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx]: Fixed

File: src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx (hunk 2 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx]: Type removed

File: src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx (hunk 3 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/ThemeSwitcher/ThemeSwitcher.tsx:15]: Aria-label added

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+          <div className="w-full pb-6 mb-6 border-b border-b-gray-200 md:pb-5 md:mb-5">
+            <Button mode="quaternaryOutline" isFullSize onClick={connectWallet}>
+              {formatMessage({ id: 'Connect wallet' })}
+            </Button>
+          </div>
+        </>
+      )}
+      <div className="w-full pb-6 mb-6 border-b border-b-gray-200 md:pb-5 md:mb-5">
+        <p className="text-xs text-gray-400 font-medium font-inter mb-3">
+          <FormattedMessage {...MSG.optionsTitle} />
```
[the remark this answers is not in the document]
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx]: formatMessage used

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+        </p>
+        <ul className="text-lg font-semibold md:font-normal md:text-md text-gray-900">
+          <li className="mb-4">
+            <Link to="/" className="flex items-center">
+              <Icon name="circles-three-plus" appearance={{ size: 'tiny' }} />
+              <p className="ml-2">
+                <FormattedMessage {...MSG.getStartedTitle} />
```
[the remark this answers is not in the document]
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx]: formatMessage ussed

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx (hunk 4 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx]: formatMessage ussed

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx (hunk 5 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx]: formatMessage ussed

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx (hunk 6 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/partials/UserMenu/UserMenu.tsx]: formatMessage ussed

File: src/components/common/Extensions/UserNavigation/partials/UserMenu/types.ts [hunk 14]
```diff
@@ -0,0 +1,10 @@
+import { PropsGetterArgs } from 'react-popper-tooltip';
+
+export interface UserMenuProps {
+  tooltipProps: (args?: PropsGetterArgs) => {
+    'data-popper-interactive': boolean | undefined;
```
[the remark this answers is not in the document]
Comment (CzarekDryl) [reply src/components/common/Extensions/UserNavigation/partials/UserMenu/types.ts]: changed to `'data-popper-interactive'?: boolean;`

**Review by KatarzynaStefanowiczPagepro — APPROVED** [review]

**Review by krawczykkatarzyna — APPROVED** [review]

**Pull request merged**